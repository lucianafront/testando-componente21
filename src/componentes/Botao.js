function Botao(props) {
  return (

    <a href={props.link}target="_blank" rel="noreferrer">
        <i className={props.ico} aria-hidden="true"></i>
    </a>

  );
}

export default Botao;
