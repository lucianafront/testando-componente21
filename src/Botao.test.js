import React, { act } from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Botao from "./componentes/Botao";

test("deve renderizar o componente Botao e verificar interações", async () => {
  const props = {
    link: "https://example.com",
    ico: "fa fa-icon",
  };

  render(<Botao {...props} />);

  // Verificar se o ícone é renderizado
  expect(screen.getByRole("link")).toBeInTheDocument();
  expect(screen.getByRole("link")).toHaveAttribute(
    "href",
    "https://example.com"
  );
  expect(screen.getByRole("link").querySelector("i")).toHaveClass("fa fa-icon");

  // Simular clique no link
  await act(() => userEvent.click(screen.getByRole("link")));

  // Verificar se o link foi clicado (pode ser necessário mockar o comportamento)
  expect(screen.getByRole("link")).toHaveAttribute(
    "href",
    "https://example.com"
  );
});

test("deve corresponder ao snapshot", () => {
  const props = {
    link: "https://example.com",
    ico: "fa fa-icon",
  };

  const { asFragment } = render(<Botao {...props} />);
  expect(asFragment()).toMatchSnapshot();
});

test("deve verificar se o componente Botao exibe o ícone correto", () => {
  const props = {
    link: "https://example.com",
    ico: "fa fa-icon",
  };

  render(<Botao {...props} />);
  expect(screen.getByRole("link").querySelector("i")).toHaveClass("fa fa-icon");
});
