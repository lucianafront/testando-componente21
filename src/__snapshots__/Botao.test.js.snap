// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deve corresponder ao snapshot 1`] = `
<DocumentFragment>
  <a
    href="https://example.com"
    rel="noreferrer"
    target="_blank"
  >
    <i
      aria-hidden="true"
      class="fa fa-icon"
    />
  </a>
</DocumentFragment>
`;
